File: Task4/main.cpp
#include <iostream>
#include <fstream>
#include <string>
#include "library.h"

int main() {
    std::ifstream inputFile("data.conf");
    if (!inputFile.is_open()) {
        std::cerr << "Не удалось открыть файл data.conf" << std::endl;
        return 1;
    }

    int num_commands;
    inputFile >> num_commands;
    inputFile.ignore(); // Игнорируем остаток строки после числа команд

    std::vector<Student> students; // Все студенты
    std::vector<int> top_list;     // Индексы студентов в списке на отчисление

    std::string line;
    for (int i = 0; i < num_commands; ++i) {
        std::getline(inputFile, line);
        process_command(line, students, top_list);
    }

    inputFile.close();
    return 0;
}

File: Task4/Java/Main.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

public class Main {

	public static void main(String[] args) {
		String filename = "data.conf";
		try (
			BufferedReader reader = new BufferedReader(new FileReader(filename))
		) {
			int numCommands = Integer.parseInt(reader.readLine());

			List<Student> students = new ArrayList<>(); // Все студенты
			List<Integer> topList = new ArrayList<>(); // ID студентов в списке на отчисление

			String line;
			for (int i = 0; i < numCommands; i++) {
				if ((line = reader.readLine()) != null) {
					Library.processCommand(line, students, topList);
				}
			}
		} catch (IOException e) {
			System.err.println(
				"Не удалось открыть файл " + filename + ": " + e.getMessage()
			);
		} catch (NumberFormatException e) {
			System.err.println(
				"Ошибка формата числа в файле data.conf: " + e.getMessage()
			);
		}
	}
}
